src/vs/workbench/browser/parts/editor/textResourceEditor.ts

```diff
@@ -191,6 +191,10 @@ export class TextResourceEditor extends AbstractTextResourceEditor {
 		// VSCode. As such we trust this languageId and use it if specific
 		if (e.languageId) {
 			candidateLanguageId = e.languageId;
+			if (this.input instanceof UntitledTextEditorInput) {
+				// High confidence guess, set the `hasLanguageSetExplicitly` flag to block future guessing
+				return this.input.model.setLanguageId(candidateLanguageId);
```

Comment:
This looks to me like the wrong location of the code and makes it hard to understand the flow. We set the mode at the end of the function, so I would move this code block also at the end so that it is clear how the mode is set (either directly or via untitled editor input).